source/common/http/conn_manager_impl.h

```diff
@@ -691,6 +696,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
   const Server::OverloadActionState& overload_stop_accepting_requests_ref_;
   const Server::OverloadActionState& overload_disable_keepalive_ref_;
   TimeSource& time_source_;
+  std::map<const std::string, PerConnectionObjectSharedPtr> per_connection_object_map_;
```

Comment:
Use absl::flat_hash_map instead

Reply:
Thanks, I am actually reading and understanding the difference in using absl::flat_hash_map against std::map.

From the stand-point of learning; could you please provide insight into this recommendation?

Is it to be compliant with the maps used in envoy; or does it perform better? 

std::map is an implementation of red-black trees (self balancing binary search trees); the reference I looked up here:
https://abseil.io/docs/cpp/guides/container

recommends these as an alternative to std::unordered_map.

I see that heterogeneous look-ups are supported on std::map; however will need to change the declartion to include `std::less<>`

I will make this change, regardless; but was just a little curious.

Thanks!
Rama

Reply:
1) You don't need an ordered map
2) flat_hash_map is faster than unordered_map

Reply:
Thanks!